File: maximum-subarray/maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        cur_arr = max_arr = nums[0]

        for num in nums[1:]:
            cur_arr = max(num, cur_arr + num)
            max_arr = max(max_arr, cur_arr)
        return max_arr

File: maximum-subarray/maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int sum = 0, maxsum = INT_MIN;
        for (int i = 0; i < nums.size(); i++) {
            sum = max(nums[i], sum + nums[i]);
            maxsum = max(sum, maxsum);
        }
        return maxsum;
    }
};
